doc/doxygen/src/advanced-tricks.md

```diff
@@ -0,0 +1,32 @@
+Advanced tricks                                              {#advanced-tricks}
```

Comment:
Since those show up in a top-level view, how about rather having

```suggestion
Advanced build system tricks                                 {#advanced-build-system-tricks}
```

Reply:
Addressed and rename the file too.